enums.rb

```diff
@@ -0,0 +1,144 @@
+module Enumerable
+  def my_each
+    for item in self do
+      yield(item)
+    end
+  end
+
+  def my_each_with_index
+    i=0
+    for item in self do
+      yield(item, i)
+      i+=1
+    end
+  end
+
+  def my_select
+    arr=[]
+      self.my_each do
+        |item| arr << item if yield(item)
+      end
+    arr
+  end
+
+  def my_all?()
+    check=false
+      self.my_each do |item|
+        check = yield(item)
+        break if not check
+      end
+    check
+  end
+
+  def my_any?
+    check = false
+      self.my_each do |item|
+        check=yield(item)
+        break if check
+      end 
+      check
+  end
+
+  def my_none?()
+    check = false
+    self.my_each do |item|
+      check = !yield(item)
+      break if not check  
+    end 
+    check
+  end
+
+  def my_count(arg=nil)
```

Comment:
please use `elsif` otherwise the last block of code would run each time, even when block not given. In fact, in line 58 it would run even when arg passed.

Consider
```ruby
if block_given?
  # do stuff
elsif arg != nil 
  # do stuff
else
  # do stuff
end
```
also perhaps there is a better way of doing line 58 to 61. simply can return the length of the original array passed (self) so basically `return length` would suffice.